internal/options/options.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2020 Red Hat, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/spf13/pflag"
+	"k8s.io/klog"
+)
+
+// Options are the configurable parameters for kube-state-metrics.
+type Options struct {
+	Host    string
+	Port    int
+	Version bool
+
+	flags *pflag.FlagSet
+}
+
+// NewOptions returns a new instance of `Options`.
+func NewOptions() *Options {
+	return &Options{}
+}
+
+// AddFlags populates the Options struct from the command line arguments passed.
+func (o *Options) AddFlags() {
+	o.flags = pflag.NewFlagSet("", pflag.ExitOnError)
+
+	// Add klog flags.
+	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+	klog.InitFlags(klogFlags)
+	o.flags.AddGoFlagSet(klogFlags)
+
+	o.flags.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+		o.flags.PrintDefaults()
+	}
+
+	o.flags.StringVar(&o.Host, "host", "0.0.0.0", `Host to expose metrics on.`)
```

Comment:
So we want to be able to host our own (exporter) metrics separately from the ones exported by this exporter. So we want to expose this on its own host and own port. So lets rename this flag to something to indicate we are configuring port for own metrics. In kube-state-metrics we call this `telemetry-port`, I would choose different name.